src/core/init/stall_avoider.ts

```diff
@@ -251,14 +251,23 @@ export default function StallAvoider(
       //                                                next period
       for (let i = manifest.periods.length - 2; i >= 0; i--) {
         const period = manifest.periods[i];
-        if (period.end !== undefined && period.end <= freezePosition) {
-          if (manifest.periods[i + 1].start > freezePosition &&
-              manifest.periods[i + 1].start > mediaElement.currentTime)
+        const contentEnd = period.end === undefined ? undefined :
```

Comment:
I don't know how to feel about that one, I liked the idea of `period.end` and `period.start` being the minimum and maximum boundaries of the Period and thus having this code just checking if we're between two periods without checking the actual content (which is already handled by the `StallAvoider` thanks to the "discontinuitiesStore$".